src/contexts/quiz/application/ports/out/questions-repo.ts

```diff
@@ -0,0 +1,7 @@
+import { QuizQuestion } from '@/contexts/quiz/entities';
+
+export interface QuestionsRepo {
+    getById(id: string): Promise<QuizQuestion>;
+    upsertQuestion(question: QuizQuestion): Promise<QuizQuestion>;
```

Comment:
i would stick with generic method names eg. `upsert` and `getByQuizId`, methods will be called in the context of repo anyway so we shouldn't introduce unnecessary repetition (`questionsRepo.upsertQuestion` vs `questionsRepo.upsert` or `repo.upsert`)